src/ucm/event/event.c

```diff
@@ -0,0 +1,422 @@
+/**
+ * Copyright (C) Mellanox Technologies Ltd. 2001-2015.  ALL RIGHTS RESERVED.
+ *
+ * See file LICENSE for terms.
+ */
+
+#ifdef HAVE_CONFIG_H
+#  include "config.h"
+#endif
+
+#include "event.h"
+
+#include <ucm/api/ucm.h>
+#include <ucm/mmap/mmap.h>
+#include <ucm/malloc/malloc_hook.h>
+#include <ucm/util/ucm_config.h>
+#include <ucm/util/log.h>
+#include <ucs/arch/cpu.h>
+
+#include <sys/mman.h>
+#include <pthread.h>
+#include <sys/shm.h>
+#include <sys/ipc.h>
+#include <stdlib.h>
+
+
+static pthread_mutex_t ucm_event_lock = PTHREAD_MUTEX_INITIALIZER;
```

Comment:
if pthread is not available many components (in ucs as well) will not work. if we want to work on such systems, need to create platform-independent locks library/wrappers in UCS.